qiskit/providers/fake_provider/__init__.py

```diff
@@ -20,23 +20,27 @@
 The mock devices are mainly for testing the compiler.
 """
 
-from qiskit.test.mock.fake_provider import (
-    FakeProviderForBackendV2,
-    FakeProvider,
-)
-from qiskit.test.mock.fake_provider import FakeProviderFactory
-from qiskit.test.mock.fake_backend import FakeBackend
-from qiskit.test.mock.fake_pulse_backend import FakePulseBackend
-from qiskit.test.mock.fake_qasm_backend import FakeQasmBackend
-from qiskit.test.mock.utils.configurable_backend import ConfigurableFakeBackend
-from qiskit.test.mock.fake_backend_v2 import FakeBackendV2, FakeBackend5QV2
```

Comment:
There is a name clash here between `fake_backend_v2.FakeBackendV2` and `fake_backend.FakeBackendV2` (part of https://github.com/Qiskit/qiskit-terra/issues/7838) so I decided not to import the classes in `qiskit.test.mock.fake_backend_v2`.